backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -818,20 +817,18 @@ IGitMacheteRepositorySnapshot discoverLayoutAndCreateSnapshot(int mostRecentlyCh
         throws GitCoreException, GitMacheteException {
```

Comment:
While we are here... are we okay with throwing these two exceptions?
Of course, this method is not being exposed outside the package but still, it is somewhat messy to throw exceptions of different layers.

Reply:
Yeah, I'm definitely not okay with that... but the alternative IIRC was that the exceptions were wrapped in method code, which would leave the code even more messy (a lot of noise added by all those extra `Try`es or `try...catch`es)